Automattic/woocommerce-payments
# Standardise Tracking Calls on the Client

Fixes #8075 

#### Changes proposed in this Pull Request





This is mainly a tech debt PR, introducing `.ts` on the tracking files and setting up a file to store the event names.





#### Testing instructions





* Run `npm run watch`.

Instructions borrowed from #6128:

> To check that Tracks events are properly registered, you can use one of the following methods:
> - Run `localStorage.setItem( 'debug', 'wc-admin:tracks' );` on your site browser console. And look for `verbose` logs.
> - Install [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) browser extension and use its log of events.
> - Go to [Tracks Live view](https://mc.a8c.com/tracks/live/), filter by any of the allowed methods, and wait patiently (~10min) for events to be registered there.
> 
> Go through the onboarding process ensuring every new track event gets registered accordingly. Apart from the description you can see [a table with screenshots here](https://docs.google.com/document/d/1XXPpXJtYPKkmOQxD43el8Mk6JTuw1IGG1SskKOvgYHE/edit).

For this PR, the desired effect is that nothing has changed - so a good way to test is to go through the onboarding flow and ensure the events are being recorded as expected - for example, we should see the following events when onboarding an account:

* `wcpay_onboarding_flow_started`
* `wcpay_onboarding_flow_mode_selected`
* `wcpay_onboarding_flow_step_completed` (for each step of the onboarding)
* `wcpay_onboarding_flow_redirected` (on Stripe redirect)

-------------------


**Post merge**
